compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1873,8 +1875,84 @@ setContractionRootConfig(mlir::FunctionOpInterface entryPointFn,
                              vecPreProcStrategy);
 }
 
+// Utility function to return the result inner tile sizes associated with the
+// mmt4d op. I.e. returns the M1 and N1 dimensions of the result shape
+// BxM0xN0xM1xN1. It also explicitly checks for scalable dimensions of the form
+// %cst * vscale, returns the constant and sets the scalable flag of the
+// corresponding dimension to true. For example, if we have the inner tile sizes
+// [8, 8*vscale], this method returns [8, 8] as tile sizes and [false, true] as
+// scalable flags.
+// TODO: By using the IterationDimTracker, extend the logic to
+// infer all inner tile sizes - including the inner K dimension, if possible.
+static std::optional<SizesAndScalableFlags>
+getMmt4dInnerTileSizes(linalg::LinalgOp op) {
+  Value destValue = op.getDpsInitOperand(0)->get();
+  ArrayRef<int64_t> destShape =
+      cast<ShapedType>(destValue.getType()).getShape();
+  // In case of static tiles, we just return the last two dim sizes.
+  if (ShapedType::isStaticShape(destShape)) {
+    return SizesAndScalableFlags{destShape.take_back(2),
+                                 IREE::Codegen::ScalableTileFlags(2, false)};
+  }
+  Operation *destOp = destValue.getDefiningOp();
+  SmallVector<OpFoldResult> innerDims;
+  // Currently, the only init tensor of the mmt4d seems to be linalg.fill ops
+  // or the bindings. Otherwise, bail out.
+  // TODO: implement ShapeAwareOpInterface for tensor.empty and track the IR
+  // until it reaches an operation that implements that interface instead of
+  // hard-coding these ops.
+  if (!isa<linalg::FillOp, IREE::TensorExt::DispatchTensorLoadOp>(destOp)) {
+    LDBG() << "Mmt4d with non-fill or tensor load init buffer, cannot infer "
+              "inner tile sizes!";
+    return std::nullopt;
+  }
+  if (auto fillOp = dyn_cast<linalg::FillOp>(destOp)) {
+    auto emptyOp = dyn_cast<tensor::EmptyOp>(
+        fillOp.getDpsInitOperand(0)->get().getDefiningOp());
+    if (!emptyOp) {
+      LDBG() << "Could not infer inner tile sizes of a scalable mmt4d op!";
+      return std::nullopt;
+    }
+    innerDims = emptyOp.getMixedSizes();
+  }
+  if (auto loadOp = dyn_cast<IREE::TensorExt::DispatchTensorLoadOp>(destOp)) {
+    innerDims = loadOp.getMixedSizes();
+  }
+  // We only need the innermost M1 and N1 dims.
+  innerDims.erase(innerDims.begin(), innerDims.end() - 2);
+  return getScalableTileSizesAndFlags(innerDims);
+}
+
+// Adjusts the tile sizes and scalable flags for SVE in-place.
+// Returns failure if it cannot adjust tile sizes.
```

Comment:
Do we update it to `does not need to adjust tile sizes` instead?

```suggestion
// Returns false if it does not need to adjust tile sizes.
```